presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergUtil.java

```diff
@@ -577,4 +584,13 @@ public static List<HivePartition> getPartitions(
 
         return new ArrayList<>(partitions);
     }
+
+    public static void loadCachingProperties(Map<String, String> properties, IcebergConfig icebergConfig)
+    {
+        properties.put(IO_MANIFEST_CACHE_ENABLED, "true");
+        properties.put(FILE_IO_IMPL, fileIOImpl);
```

Comment:
@agrawalreetika Do you think what @nastra suggested make sense? If yes could you please update the code?

Reply:
@nastra Currently fileIOImpl is being picked up from IcebergConfig which is marked as NotNull currently. We can make this optional once we upgrade Iceberg to version which has changes from https://github.com/apache/iceberg/pull/9283

Reply:
Yes, this should be optional, because it's also an optional configuration in Iceberg itself. Also I still think it `FILE_IO_IMPL` should be configured outside of the `loadCachingProperties()` method (independently from whether it's optional/required right now)

Reply:
Sure, I have taken it out from loadCachingProperties() method